Infrastructure/Services/ProcessEmailsService.cs

```diff
@@ -0,0 +1,450 @@
+        }
+        else if (!string.IsNullOrWhiteSpace(emailToBeUsedIfTestingEnabled) && _emailSettingsConfig.AllSentToEnquirer)
+        {
+            testingEmail = emailToBeUsedIfTestingEnabled;
+        }
+
+        return testingEmail;
+    }
+
+    public EmailLog? MergeEmailForTesting(List<EmailLog> emailLogs, List<string> personalisationPropertiesToMerge)
+    {
+        if (_emailSettingsConfig.MergeResponses &&
+            emailLogs.Count > 1 &&
+            emailLogs.First().EmailPersonalisationLogs != null &&
+            personalisationPropertiesToMerge.Count > 0)
+        {
+            var initialEmailLog = emailLogs.First();
+            var initialEmailPersonalisationLogs = initialEmailLog!.EmailPersonalisationLogs!;
+
+            foreach (var personalisationPropertyToMerge in personalisationPropertiesToMerge)
+            {
+                var value = string.Empty;
+
+                var addNewLine = string.Empty;
+
+                foreach (var emailLog in emailLogs)
+                {
+                    var matchedEmailPersonalisationLog = emailLog!.EmailPersonalisationLogs!.Single(x => x.Key.Equals(personalisationPropertyToMerge));
+                    value = $"{value}{addNewLine}{emailLog.EmailAddress} - {matchedEmailPersonalisationLog.Value}";
+                    addNewLine = $"{Environment.NewLine}{Environment.NewLine}";
+                }
+
+                var initialEmailPersonalisationLog = initialEmailPersonalisationLogs.Single(x => x.Key.Equals(personalisationPropertyToMerge));
+                initialEmailPersonalisationLog.Value = value;
+            }
+
+            initialEmailPersonalisationLogs.Add(new EmailPersonalisationLog()
+            {
+                Key = TestExtraInfoKey,
+                Value = $"This is a merged email for testing purposes.{Environment.NewLine}{Environment.NewLine}"
+            });
+
+            initialEmailLog.EmailAddress = "merged_email_for_testing@education.gov.uk";
+
+            return initialEmailLog;
+        }
+
+        return null;
+    }
+
+    public bool SendTuitionPartnerEmailsWhenEnquirerDelivered()
+    {
+        return _emailSettingsConfig.SendTuitionPartnerEmailsWhenEnquirerDelivered;
+    }
+
+    private async Task<bool> StartProcessing(DateTime date)
+    {
+        var scheduledProcessingInfo = await _unitOfWork.ScheduledProcessingInfoRepository
+            .SingleOrDefaultAsync(x => x.ScheduleName == ScheduleName);
+
+        if (scheduledProcessingInfo == null)
+        {
+            _unitOfWork.ScheduledProcessingInfoRepository.AddAsync(new ScheduledProcessingInfo()
+            {
+                ScheduleName = ScheduleName,
+                LastStartedDate = date
+            });
+
+            await _unitOfWork.Complete();
+        }
+        else
+        {
+            if (scheduledProcessingInfo.LastFinishedDate == null)
+            {
+                if (scheduledProcessingInfo.LastStartedDate.AddMinutes(30) < DateTime.UtcNow)
```

Comment:
The code contains the magic number `30`. Consider using constants value to give this number a meaningful name.

Reply:
Done